### ISSUE TITLE
[Bug]: Click ImportError due to Typer dependency

### ISSUE THREAD

**Issue by davebulaval** [rstcheck/rstcheck#138]
### To Dos

- [X] I tested with the [latest released version](https://github.com/rstcheck/rstcheck/releases/latest).

- [X] I have checked the [issues](https://github.com/rstcheck/rstcheck/issues) and think that this is not a duplicate.

- [X] I added a very descriptive title to this issue.

### Example Code (rst)

```rst
`rstcheck docs/source/examples/fine_tuning.rst`
```

### Relevant log output

```Shell
Traceback (most recent call last):
  File "/home/davebulaval/anaconda3/bin/rstcheck", line 5, in <module>
    from rstcheck._cli import main
  File "/home/davebulaval/anaconda3/lib/python3.9/site-packages/rstcheck/_cli.py", line 6, in <module>
    import typer
  File "/home/davebulaval/anaconda3/lib/python3.9/site-packages/typer/__init__.py", line 12, in <module>
    from click.termui import get_terminal_size as get_terminal_size
ImportError: cannot import name 'get_terminal_size' from 'click.termui' (/home/davebulaval/anaconda3/lib/python3.9/site-packages/click/termui.py)

```

### Description

Typer has a running issue with is Click dependency that raises an ImportError.

By using a Click version greater than 8.0.4, it raises an ImportError.

A quick workaround is to set Click to 8.0.4 as discussed in [Typer issue](https://github.com/tiangolo/typer/issues/377).

I post this either as a documented solution for the problem or as a discussion to fix dependencies in this project

### Operating System

Linux

### Python Version

3.9

### rstcheck Version

6.0.0.post1

EDIT by @Cielquan: removed redundant code block; fix version string

**Comment by Cielquan** [comment]
If you look at your linked issue's linked PRs there is a closed one for click 8.1 support. It was closed in favor of another PR, which implemented the support too. The latter PR was released with typer version 0.4.1.

Source: https://github.com/tiangolo/typer/pull/375#issuecomment-1083536384

The current latest typer version is already 0.6.1. I would lift the lower bound of typer rather than lowering the upper bound of click as a more future oriented approach. But would also add a note to the docs.

I will make this quick change and some other tooling version bumping and plan to release this under 6.1.0 really soon. Probably even today.

**Comment by Cielquan** [comment]
Fix is released with 6.1

**Comment by davebulaval** [comment]
Awesome! Tks for the quick release.
